# Topic: LEVERAGED AND INVERSE ETPs

# Course: SPRING 2023 PROJECT REPORT (GFGB-8951-002)

# Advisor: PROF. YUEWU XU

# Student: ATAKAN DEMIRCI

# FIDN: A18767841

## Summary

The paper by Colby J. Pessina and Robert E. Whaley discusses leveraged and inverse exchange-traded products (ETPs) and examines their value-destroying characteristics, which are not well understood by investors. Leveraged and inverse ETPs offer an inexpensive, convenient, highly levered, and limited-liability means for profiting from a directional price view. However, these products are not suitable for buy-and-hold investments or effective hedging tools. Trading costs and barriers to entry are low, and the most significant problem with leveraged and inverse funds is that most of them are expected to collapse. The paper concludes that leveraged and inverse ETPs are not effective investment management tools. We wanted to extend the study to 2023 with publicly available data of S&P 500 and Russell 2000.


### Background

"Levered and Inverse Exchange-Traded Products: Blessing or Curse?" by Colby J. Pessina & Robert E. Whaley states that Levered and inverse exchange-trade products (ETPs) are designed to provide geared long and short exposures to the daily returns of various benchmark indexes. The benchmarks may be any reference index, but the popular ones are indexes of stocks, bonds, commodities, and volatility. The problem with these products is that they are not generally well understood, particularly those with futures-based benchmarks. Levered and inverse ETPs are neither suitable buy-and-hold investments nor effective hedging tools. They are unstable and exist only as mechanisms for placing short-term directional bets. Levered and inverse products are not, and cannot be effective investment management tools.

In [1]:
# Here we start with the libraries we will need
import os
import yfinance as yf
import pandas as pd
import numpy as np
import statsmodels.api as sm

# Download the Data

### Benchmark Data
We start with our benchmark data. Only challenge s far is 20 year T-Bond data. We will get to that later

In [69]:
# Download data for S&P 500
spx_data = yf.download("^GSPC", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(spx_data.head())

# Save the data to a CSV file
spx_data.to_csv("spx.csv")


[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  \
Date                                                                     
2009-06-26  918.840027  922.000000  913.030029  918.900024  918.900024   
2009-06-29  919.859985  927.989990  916.179993  927.229980  927.229980   
2009-06-30  927.150024  930.010010  912.859985  919.320007  919.320007   
2009-07-01  920.820007  931.919983  920.820007  923.330017  923.330017   
2009-07-02  921.239990  921.239990  896.419983  896.419983  896.419983   

                Volume  
Date                    
2009-06-26  6076660000  
2009-06-29  4211760000  
2009-06-30  4627570000  
2009-07-01  3919400000  
2009-07-02  3931000000  


In [38]:
# Download data for Russell 2000
rut_data = yf.download("^RUT", start="2008-11-19", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(rut_data.head())

# Save the data to a CSV file
rut_data.to_csv("rut.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  \
Date                                                                     
2008-11-19  446.260010  448.809998  412.380005  412.380005  412.380005   
2008-11-20  411.660004  417.350006  384.579987  385.309998  385.309998   
2008-11-21  389.579987  406.750000  371.299988  406.540009  406.540009   
2008-11-24  407.459991  438.980011  407.459991  436.799988  436.799988   
2008-11-25  438.779999  443.179993  426.709991  443.179993  443.179993   

                Volume  
Date                    
2008-11-19  6548600000  
2008-11-20  9093740000  
2008-11-21  9495900000  
2008-11-24  7879440000  
2008-11-25  6952700000  


In [3]:
# Download data for IDCOT20TR
idcot20tr_data = yf.download("^IDCOT20TR", start="2005-12-19", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(idcot20tr_data.head())

# Save the data to a CSV file
idcot20tr_data.to_csv("idcot20tr.csv")

[*********************100%***********************]  1 of 1 completed

1 Failed download:
- ^IDCOT20TR: No data found for this date range, symbol may be delisted
Empty DataFrame
Columns: [Open, High, Low, Close, Adj Close, Volume]
Index: []


### S&P 500 Levered and Reversed ETPs


In [2]:
# Download data for UPRO
upro_data = yf.download("UPRO", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(upro_data.head())

# Save the data to a CSV file
upro_data.to_csv("upro.csv")

[*********************100%***********************]  1 of 1 completed
                Open      High       Low     Close  Adj Close    Volume
Date                                                                   
2009-06-26  1.195278  1.213056  1.177361  1.199306   1.159664  13104000
2009-06-29  1.208333  1.236111  1.191389  1.233333   1.192567   8690400
2009-06-30  1.233333  1.243333  1.176389  1.203889   1.164096  17128800
2009-07-01  1.218472  1.252917  1.213611  1.216806   1.176586  12038400
2009-07-02  1.169306  1.169861  1.115972  1.116389   1.079488  29275200


In [3]:
# Download data for SPXS
spxs_data = yf.download("SPXS", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(spxs_data.head())

# Save the data to a CSV file
spxs_data.to_csv("spxs.csv")

[*********************100%***********************]  1 of 1 completed
               Open     High      Low    Close     Adj Close  Volume
Date                                                                
2009-06-26  44575.0  45162.5  43712.5  44325.0  42931.878906    6742
2009-06-29  43875.0  44687.5  42962.5  43125.0  41769.589844    6759
2009-06-30  43162.5  44987.5  42600.0  43937.5  42556.554688    9328
2009-07-01  43462.5  43575.0  42200.0  43462.5  42096.484375    8685
2009-07-02  45175.0  47000.0  45112.5  46950.0  45474.371094   10179


In [4]:
# Download data for SPXU
spxu_data = yf.download("SPXU", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(spxu_data.head())

# Save the data to a CSV file
spxu_data.to_csv("spxu.csv")

[*********************100%***********************]  1 of 1 completed
               Open     High      Low    Close     Adj Close  Volume
Date                                                                
2009-06-26  29628.0  30016.0  29148.0  29532.0  28186.097656     315
2009-06-29  29280.0  29664.0  28600.0  28652.0  27346.199219     332
2009-06-30  28684.0  30008.0  28372.0  29352.0  28014.300781     763
2009-07-01  29000.0  29100.0  28172.0  29044.0  27720.333984     797
2009-07-02  30168.0  31444.0  30152.0  31360.0  29930.789062     721


In [5]:
# Download data for SDS
sds_data = yf.download("SDS", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(sds_data.head())

# Save the data to a CSV file
sds_data.to_csv("sds.csv")

[*********************100%***********************]  1 of 1 completed
                   Open         High          Low        Close    Adj Close  \
Date                                                                          
2009-06-26  4444.799805  4492.000000  4400.000000  4443.200195  4259.951172   
2009-06-29  4412.799805  4460.799805  4345.600098  4352.799805  4173.279785   
2009-06-30  4351.200195  4488.799805  4324.000000  4426.399902  4243.843262   
2009-07-01  4389.600098  4396.799805  4303.200195  4388.000000  4207.027344   
2009-07-02  4504.000000  4632.000000  4502.399902  4627.200195  4436.363281   

            Volume  
Date                
2009-06-26  409653  
2009-06-29  388384  
2009-06-30  444098  
2009-07-01  426953  
2009-07-02  439908  


In [6]:
# Download data for SH
sh_data = yf.download("SH", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(sh_data.head())

# Save the data to a CSV file
sh_data.to_csv("sh.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  \
Date                                                                     
2009-06-26  131.619995  132.339996  131.000000  131.619995  126.523232   
2009-06-29  131.179993  131.860001  130.119995  130.320007  125.273582   
2009-06-30  130.179993  132.300003  129.800003  131.419998  126.330994   
2009-07-01  130.699997  130.919998  129.500000  130.839996  125.773445   
2009-07-02  132.559998  134.440002  132.500000  134.419998  129.214798   

             Volume  
Date                 
2009-06-26   883150  
2009-06-29   600550  
2009-06-30   878650  
2009-07-01   758700  
2009-07-02  1384300  


In [7]:
# Download data for SSO
sso_data = yf.download("SSO", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(sso_data.head())

# Save the data to a CSV file
sso_data.to_csv("sso.csv")

[*********************100%***********************]  1 of 1 completed
               Open     High      Low    Close  Adj Close     Volume
Date                                                                
2009-06-26  3.25000  3.28000  3.21375  3.25250   3.041738  249833600
2009-06-29  3.27250  3.32125  3.23625  3.31375   3.099017  255537600
2009-06-30  3.31250  3.33750  3.21375  3.26750   3.055765  309218400
2009-07-01  3.28750  3.35000  3.27750  3.28375   3.070961  206745600
2009-07-02  3.20125  3.20250  3.10375  3.10750   2.906133  293230400


In [8]:
# Download data for SPXL
spxl_data = yf.download("SPXL", start="2009-06-26", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(spxl_data.head())

# Save the data to a CSV file
spxl_data.to_csv("spxl.csv")

[*********************100%***********************]  1 of 1 completed
                Open      High       Low     Close  Adj Close     Volume
Date                                                                    
2009-06-26  2.795000  2.849167  2.762500  2.807500   2.246401  133370400
2009-06-29  2.841667  2.900833  2.791667  2.887500   2.310412  113755200
2009-06-30  2.887500  2.923333  2.763333  2.830000   2.264404  179834400
2009-07-01  2.864167  2.941667  2.850000  2.862500   2.290409  120717600
2009-07-02  2.748333  2.749167  2.625000  2.626667   2.101709  146991600


### Russell 2000 ETPs

In [10]:
# Download data for TZA
tza_data = yf.download("TZA", start="2008-11-06", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tza_data.head())

# Save the data to a CSV file
tza_data.to_csv("tza.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close    Adj Close  Volume
Date                                                                       
2008-11-19   917088.0  1130304.0   891168.0  1124256.0  1085505.875      84
2008-11-20  1156320.0  1360608.0  1074720.0  1349184.0  1302681.125     142
2008-11-21  1257600.0  1503168.0  1113984.0  1125312.0  1086525.375     175
2008-11-24  1068576.0  1123392.0   848160.0   885984.0   855446.375     107
2008-11-25   846624.0   939168.0   833088.0   843936.0   814847.750     109


##### Note:
Yahoo warns us historical price date shouldn't be prior to 2008-11-19. We match our data according to this.

In [52]:
# Download data for TWN
twm_data = yf.download("TWM", start="2008-11-19", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(twm_data.head())

# Save the data to a CSV file
twm_data.to_csv("twm.csv")

[*********************100%***********************]  1 of 1 completed
                    Open          High          Low         Close  \
Date                                                                
2008-11-19   8225.919922   9515.519531  8090.240234   9489.919922   
2008-11-20   9630.719727  10787.839844  9201.919922  10745.599609   
2008-11-21  10345.599609  11560.959961  9488.000000   9597.440430   
2008-11-24   9337.599609   9546.879883  7998.720215   8256.000000   
2008-11-25   7960.959961   8540.799805  7872.640137   7954.560059   

              Adj Close  Volume  
Date                             
2008-11-19  6763.792969  135656  
2008-11-20  7658.759277  156584  
2008-11-21  6840.425781  153614  
2008-11-24  5884.335938  117295  
2008-11-25  5669.488281   92897  


In [41]:
# Download data for RWM
rwm_data = yf.download("RWM", start="2008-11-19", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(rwm_data.head())

# Save the data to a CSV file
rwm_data.to_csv("rwm.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  Volume
Date                                                                          
2008-11-19  423.200012  456.160004  420.239990  455.679993  324.349640   23325
2008-11-20  459.959991  488.000000  449.160004  485.720001  345.731934   32100
2008-11-21  476.880005  504.480011  457.559998  458.839996  326.598969  134325
2008-11-24  452.399994  456.799988  421.959991  427.760010  304.476440   42400
2008-11-25  420.480011  435.119995  419.799988  420.399994  299.237610  111025


In [42]:
# Download data for UWM
uwm_data = yf.download("UWM", start="2008-11-19", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(uwm_data.head())

# Save the data to a CSV file
uwm_data.to_csv("uwm.csv")

[*********************100%***********************]  1 of 1 completed
              Open   High     Low   Close  Adj Close    Volume
Date                                                          
2008-11-19  4.3325  4.415  3.6650  3.6650   3.559508  43526800
2008-11-20  3.6125  3.785  3.1525  3.1525   3.061760  47731600
2008-11-21  3.3250  3.560  2.9475  3.5325   3.430821  46645600
2008-11-24  3.6275  4.110  3.5425  3.9950   3.880009  35409200
2008-11-25  4.1425  4.175  3.8550  4.1375   4.018407  29933200


In [14]:
# Download data for TNA
tna_data = yf.download("TNA", start="2008-11-06", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tna_data.head())

# Save the data to a CSV file
tna_data.to_csv("tna.csv")

[*********************100%***********************]  1 of 1 completed
              Open    High     Low   Close  Adj Close    Volume
Date                                                           
2008-11-19  7.7675  7.9975  6.0000  6.0575   5.108106  10955600
2008-11-20  5.8350  6.2750  4.7500  4.7850   4.035046  19968400
2008-11-21  5.5475  5.6175  4.2675  5.5275   4.661174  19014400
2008-11-24  5.8725  6.9300  5.6000  6.6800   5.633042  14030400
2008-11-25  7.0100  7.1100  6.2775  7.0025   5.904996  11441200


##### Note:
TNA starts from November 19th as well

### 20 Year T-Bond

In [15]:
# Download data for TMV
tmv_data = yf.download("TMV", start="2010-01-22", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tmv_data.head())

# Save the data to a CSV file
tmv_data.to_csv("tmv.csv")

[*********************100%***********************]  1 of 1 completed
              Open    High     Low   Close    Adj Close  Volume
Date                                                           
2010-01-22  3139.5  3167.0  3111.5  3136.5  3029.329102    1432
2010-01-25  3191.0  3198.5  3148.5  3186.0  3077.137695    1200
2010-01-26  3132.5  3208.0  3132.0  3186.5  3077.620605    1176
2010-01-27  3169.5  3203.5  3133.5  3178.0  3069.410889    1796
2010-01-28  3233.0  3241.0  3178.5  3203.5  3094.039795    1830


In [16]:
# Download data for TBT
tbt_data = yf.download("TBT", start="2010-01-22", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tbt_data.head())

# Save the data to a CSV file
tbt_data.to_csv("tbt.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  \
Date                                                                     
2010-01-22  189.320007  190.639999  188.440002  189.399994  180.377808   
2010-01-25  191.600006  191.800003  189.960007  191.320007  182.206360   
2010-01-26  189.360001  192.360001  189.000000  191.000000  181.901581   
2010-01-27  190.639999  191.960007  189.080002  191.080002  181.977798   
2010-01-28  192.839996  193.240005  190.520004  191.160004  182.054001   

             Volume  
Date                 
2010-01-22  1784150  
2010-01-25  1201650  
2010-01-26  1181575  
2010-01-27  1337975  
2010-01-28  1471725  


In [17]:
# Download data for TBF
tbf_data = yf.download("TBF", start="2010-01-22", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tbf_data.head())

# Save the data to a CSV file
tbf_data.to_csv("tbf.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close  Adj Close  Volume
Date                                                                     
2010-01-22  49.230000  49.330002  49.070000  49.209999  47.255592  283400
2010-01-25  49.480000  49.490002  49.250000  49.410000  47.447651  217000
2010-01-26  49.180000  49.520000  49.119999  49.400002  47.438046   73800
2010-01-27  49.340000  49.450001  49.139999  49.400002  47.438046  119500
2010-01-28  49.720001  49.720001  49.340000  49.419998  47.457249  251700


In [18]:
# Download data for UBT
ubt_data = yf.download("UBT", start="2010-01-22", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(ubt_data.head())

# Save the data to a CSV file
ubt_data.to_csv("ubt.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close  Adj Close  Volume
Date                                                                     
2010-01-22  18.007500  18.072500  17.889999  17.990000  16.249453  257600
2010-01-25  17.817499  17.927500  17.797501  17.799999  16.077835  228000
2010-01-26  17.825001  17.852501  17.715000  17.730000  16.014608   38800
2010-01-27  17.862499  18.004999  17.742500  17.832500  16.107193   96000
2010-01-28  17.639999  17.844999  17.637501  17.764999  16.046227  358800


In [19]:
# Download data for TMF
tmf_data = yf.download("TMF", start="2010-01-22", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tmf_data.head())

# Save the data to a CSV file
tmf_data.to_csv("tmf.csv")

[*********************100%***********************]  1 of 1 completed
              Open    High     Low   Close  Adj Close  Volume
Date                                                         
2010-01-22  8.3500  8.4025  8.2550  8.3425   7.158132  118400
2010-01-25  8.1875  8.2650  8.1875  8.2025   7.038008   70400
2010-01-26  8.3525  8.3525  8.1400  8.2100   7.044441   81600
2010-01-27  8.2225  8.3325  8.1500  8.2225   7.055169  107200
2010-01-28  8.0650  8.1900  8.0575  8.1375   6.982235  130400


### Crude Oil

##### Note:

On March 19, 2020 Citi decided to discontinue DWT and UWT ETNs

In [20]:
# Download data for OILD
oild_data = yf.download("OILD", start="2017-03-28", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(oild_data.head())

# Save the data to a CSV file
oild_data.to_csv("oild.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  Volume
Date                                                                          
2021-11-09  253.399994  257.500000  250.100006  250.100006  250.100006     120
2021-11-10  252.300003  274.700012  250.699997  270.980011  270.980011     130
2021-11-11  263.000000  267.000000  263.000000  267.000000  267.000000      30
2021-11-12  269.500000  269.500000  268.529999  268.529999  268.529999      20
2021-11-15  266.850006  266.850006  262.820007  262.820007  262.820007      10


##### Note:
OILD data starts from 2021

In [21]:
# Download data for SCO
sco_data = yf.download("SCO", start="2017-03-28", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(sco_data.head())

# Save the data to a CSV file
sco_data.to_csv("sco.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  Volume
Date                                                                          
2017-03-28  812.000000  815.599976  792.599976  806.200012  806.200012  117420
2017-03-29  801.799988  801.799988  766.599976  770.599976  770.599976  128675
2017-03-30  755.200012  760.400024  738.000000  740.799988  740.799988  129280
2017-03-31  748.400024  748.799988  728.400024  731.400024  731.400024   82415
2017-04-03  734.400024  747.400024  732.599976  743.599976  743.599976   58080


In [22]:
# Download data for USO
uso_data = yf.download("USO", start="2017-03-28", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(uso_data.head())

# Save the data to a CSV file
uso_data.to_csv("uso.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close  Adj Close   Volume
Date                                                                      
2017-03-28  80.879997  81.839996  80.720001  81.199997  81.199997  2226100
2017-03-29  81.440002  83.199997  81.360001  82.959999  82.959999  2894438
2017-03-30  83.760002  84.720001  83.519997  84.559998  84.559998  3571350
2017-03-31  84.160004  85.279999  84.080002  85.120003  85.120003  2428825
2017-04-03  84.959999  85.040001  84.160004  84.400002  84.400002  2099513


In [23]:
# Download data for UCO
uco_data = yf.download("UCO", start="2017-03-28", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(uco_data.head())

# Save the data to a CSV file
uco_data.to_csv("uco.csv")

[*********************100%***********************]  1 of 1 completed
                Open      High       Low     Close  Adj Close   Volume
Date                                                                  
2017-03-28  107.6250  110.0000  107.0625  108.3125   108.3125   837984
2017-03-29  108.9375  113.5625  108.8750  112.8750   112.8750  1172736
2017-03-30  115.2500  117.8125  114.5000  117.2500   117.2500  1302976
2017-03-31  116.1250  119.3750  116.0000  118.8750   118.8750   961168
2017-04-03  118.2500  118.6875  116.1875  116.8750   116.8750   664800


In [24]:
# Download data for OILU
oilu_data = yf.download("OILU", start="2017-03-28", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(oilu_data.head())

# Save the data to a CSV file
oilu_data.to_csv("oilu.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close  Adj Close  Volume
Date                                                                     
2021-11-09  24.959999  25.290001  24.139999  25.290001  25.290001    1500
2021-11-10  24.799999  24.799999  22.559999  22.957001  22.957001    5600
2021-11-11  23.610001  23.622999  23.271000  23.271000  23.271000    2100
2021-11-12  22.910000  23.122999  22.910000  23.122999  23.122999    2600
2021-11-15  22.770000  23.573999  22.770000  23.573999  23.573999    2000


##### Note:
Data starts from 2021

### Natural Gas

##### Note:

Credit Suisse delisted UGAZ and DGAZ on July 12, 2020

In [25]:
# Download data for KOLD
kold_data = yf.download("KOLD", start="2012-02-08", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(kold_data.head())

# Save the data to a CSV file
kold_data.to_csv("kold.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  Volume
Date                                                                          
2012-02-08  830.666687  843.288879  791.466675  841.688904  841.688904    2205
2012-02-09  825.333313  855.555542  751.377808  820.444458  820.444458    4129
2012-02-10  805.066650  824.355530  800.088867  804.266663  804.266663     709
2012-02-13  834.222229  837.333313  812.622192  827.288879  827.288879    2554
2012-02-14  790.933350  798.755554  773.422241  784.444458  784.444458    1575


In [26]:
# Download data for UNG
ung_data = yf.download("UNG", start="2012-02-08", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(ung_data.head())

# Save the data to a CSV file
ung_data.to_csv("ung.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close  Adj Close   Volume
Date                                                                      
2012-02-08  83.839996  85.279999  83.040001  83.040001  83.040001   951113
2012-02-09  84.000000  88.000000  81.919998  83.839996  83.839996  2622913
2012-02-10  84.959999  84.959999  83.360001  84.320000  84.320000   761713
2012-02-13  82.559998  84.160004  82.239998  82.879997  82.879997  1106313
2012-02-14  85.599998  86.879997  85.120003  86.080002  86.080002  1719563


In [27]:
# Download data for BOIL
boil_data = yf.download("BOIL", start="2012-02-08", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(boil_data.head())

# Save the data to a CSV file
boil_data.to_csv("boil.csv")

[*********************100%***********************]  1 of 1 completed
               Open     High      Low    Close  Adj Close  Volume
Date                                                             
2012-02-08  12950.0  13370.0  12720.0  12750.0    12750.0     602
2012-02-09  13000.0  14140.0  12450.0  13060.0    13060.0     891
2012-02-10  13300.0  13430.0  12990.0  13310.0    13310.0     307
2012-02-13  12800.0  13200.0  12760.0  12960.0    12960.0     549
2012-02-14  13520.0  13750.0  13330.0  13600.0    13600.0     416


### Volatility

In [29]:
# Download data for SVXY
svxy_data = yf.download("SVXY", start="2011-10-05", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(svxy_data.head())

# Save the data to a CSV file
svxy_data.to_csv("svxy.csv")

[*********************100%***********************]  1 of 1 completed
                 Open       High        Low      Close  Adj Close  Volume
Date                                                                     
2011-10-05  21.764999  22.820000  21.725000  22.695000  22.695000   17800
2011-10-06  22.715000  23.165001  22.400000  23.165001  23.165001   11200
2011-10-07  23.594999  23.594999  22.389999  23.344999  23.344999   36400
2011-10-10  23.750000  24.299999  23.700001  24.299999  24.299999   30000
2011-10-11  24.125000  24.825001  24.120001  24.629999  24.629999   32800


In [30]:
# Download data for VXX
vxx_data = yf.download("VXX", start="2011-10-05", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(vxx_data.head())

# Save the data to a CSV file
vxx_data.to_csv("vxx.csv")

[*********************100%***********************]  1 of 1 completed
                  Open        High         Low       Close   Adj Close  Volume
Date                                                                          
2018-01-25  442.559998  442.559998  442.559998  442.559998  442.559998       0
2018-01-26  442.559998  442.559998  442.559998  442.559998  442.559998       0
2018-01-29  467.200012  473.279999  467.200012  473.279999  473.279999      88
2018-01-30  492.640015  508.959991  487.839996  488.799988  488.799988    3956
2018-01-31  479.359985  490.880005  479.359985  490.399994  490.399994     575


##### Note:
VXX data starts from 2018 January 25

In [31]:
# Download data for VIXY
vixy_data = yf.download("VIXY", start="2011-10-05", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(vixy_data.head())

# Save the data to a CSV file
vixy_data.to_csv("vixy.csv")

[*********************100%***********************]  1 of 1 completed
               Open     High      Low    Close  Adj Close  Volume
Date                                                             
2011-10-05  44696.0  45448.0  42888.0  43064.0    43064.0     904
2011-10-06  43296.0  44024.0  42280.0  42344.0    42344.0     483
2011-10-07  41520.0  43840.0  41488.0  43112.0    43112.0    1126
2011-10-10  41320.0  41400.0  39968.0  39984.0    39984.0     476
2011-10-11  40472.0  40472.0  39220.0  39508.0    39508.0     321


In [32]:
# Download data for TVIX
tvix_data = yf.download("TVIX", start="2011-10-05", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(tvix_data.head())

# Save the data to a CSV file
tvix_data.to_csv("tvix.csv")

[*********************100%***********************]  1 of 1 completed
                    Open          High           Low         Close  \
Date                                                                 
2011-10-05  2.098500e+09  2.184170e+09  1.939000e+09  1.950250e+09   
2011-10-06  1.960750e+09  2.040100e+09  1.880750e+09  1.890000e+09   
2011-10-07  1.811500e+09  2.023000e+09  1.809500e+09  1.947500e+09   
2011-10-10  1.798500e+09  1.803000e+09  1.675000e+09  1.675000e+09   
2011-10-11  1.711250e+09  1.715750e+09  1.609000e+09  1.627750e+09   

               Adj Close  Volume  
Date                              
2011-10-05  1.950250e+09       0  
2011-10-06  1.890000e+09       0  
2011-10-07  1.947500e+09       0  
2011-10-10  1.675000e+09       0  
2011-10-11  1.627750e+09       0  


In [33]:
# Download data for UVXY
uvxy_data = yf.download("UVXY", start="2011-10-05", end="2023-03-31", interval="1d")

# Print the first few rows of the data
print(uvxy_data.head())

# Save the data to a CSV file
uvxy_data.to_csv("uvxy.csv")

[*********************100%***********************]  1 of 1 completed
                    Open          High           Low         Close  \
Date                                                                 
2011-10-05  1.974600e+09  1.989000e+09  1.813800e+09  1.813800e+09   
2011-10-06  1.811400e+09  1.878000e+09  1.740000e+09  1.740000e+09   
2011-10-07  1.705800e+09  1.857600e+09  1.684800e+09  1.765200e+09   
2011-10-10  1.654800e+09  1.656000e+09  1.559400e+09  1.559400e+09   
2011-10-11  1.570200e+09  1.572600e+09  1.491000e+09  1.506000e+09   

               Adj Close  Volume  
Date                              
2011-10-05  1.813800e+09       0  
2011-10-06  1.740000e+09       0  
2011-10-07  1.765200e+09       0  
2011-10-10  1.559400e+09       0  
2011-10-11  1.506000e+09       0  


# Create Pandas DataFrames of downloaded .csv flies

We also add return and log_return columns to our tables because we ar going to need them when we calculate Tracking Error

In [53]:
# Define the directory where the CSV files are located
csv_dir = 'C:/Users/Atakan/Documents/Project_ETPs'

# Loop through the CSV files in the directory
for filename in os.listdir(csv_dir):
    if filename.endswith('.csv'):
        # Read the CSV file into a DataFrame
        filepath = os.path.join(csv_dir, filename)
        df = pd.read_csv(filepath, index_col=0)
        
        # Calculate returns as the percentage change in price from one row to the next
        df['Returns'] = df['Adj Close'].pct_change()
        
        # Calculate log returns as the log difference in price from one row to the next
        df['Log Returns'] = np.log(df['Adj Close']).diff()
        
        # drop rows with NaN values in column 'Returns'
        df.dropna(subset=['Returns'], inplace=True)
        
        # Define the name of the DataFrame
        df_name = os.path.splitext(filename)[0]
        
        # Assign the DataFrame to a variable with the appropriate name
        exec('{} = df'.format(df_name))
        
        # Print the first few rows of the DataFrame
        print(df_name)
        print(df.head())

boil
               Open     High      Low    Close  Adj Close  Volume   Returns  \
Date                                                                          
2012-02-09  13000.0  14140.0  12450.0  13060.0    13060.0     891  0.024314   
2012-02-10  13300.0  13430.0  12990.0  13310.0    13310.0     307  0.019142   
2012-02-13  12800.0  13200.0  12760.0  12960.0    12960.0     549 -0.026296   
2012-02-14  13520.0  13750.0  13330.0  13600.0    13600.0     416  0.049383   
2012-02-15  13120.0  13200.0  12500.0  12860.0    12860.0     674 -0.054412   

            Log Returns  
Date                     
2012-02-09     0.024023  
2012-02-10     0.018962  
2012-02-13    -0.026648  
2012-02-14     0.048202  
2012-02-15    -0.055948  
idcot20tr
Empty DataFrame
Columns: [Open, High, Low, Close, Adj Close, Volume, Returns, Log Returns]
Index: []
kold
                  Open        High         Low       Close   Adj Close  \
Date                                                                 

tna
              Open    High     Low   Close  Adj Close    Volume   Returns  \
Date                                                                        
2008-11-20  5.8350  6.2750  4.7500  4.7850   4.035046  19968400 -0.210070   
2008-11-21  5.5475  5.6175  4.2675  5.5275   4.661174  19014400  0.155173   
2008-11-24  5.8725  6.9300  5.6000  6.6800   5.633042  14030400  0.208503   
2008-11-25  7.0100  7.1100  6.2775  7.0025   5.904996  11441200  0.048278   
2008-11-26  6.5700  8.3175  6.5175  8.2475   6.954867  14252400  0.177794   

            Log Returns  
Date                     
2008-11-20    -0.235811  
2008-11-21     0.144250  
2008-11-24     0.189382  
2008-11-25     0.047149  
2008-11-26     0.163643  
tvix
                    Open          High           Low         Close  \
Date                                                                 
2011-10-06  1.960750e+09  2.040100e+09  1.880750e+09  1.890000e+09   
2011-10-07  1.811500e+09  2.023000e+09  1.809500e+09  1.947

## Create Panels 

### S&P

We are creating the panels of data where we compare funds to their benchmarks. 

We need to know how many observations on benchmark and Benchmark Holding Period Return (HPR)

Let's start with S&P 500

In [3]:
spx.shape

(3463, 8)

In [4]:
start_value = spx.iloc[0]['Adj Close']
end_value = spx.iloc[-1]['Adj Close']

benchmark_HPR = (end_value/start_value)-1
benchmark_HPR

3.368743637988443

S&P 500 Observations: 3463 and HPR is 336.87% during the period of June 26, 2009 - March 31, 2023

Let's create an emty Pandas dataframe for our panel starting from inverse -3 to levereged 3 funds top to bottom respectively

In [5]:
spx_fund_names = ['spxs', 'spxu', 'sds', 'sh', 'sso', 'spxl', 'upro']

spx_te_df = pd.DataFrame(columns=[])
spx_te_df['funds'] = spx_fund_names
spx_te_df= spx_te_df.set_index('funds')
spx_te_df

Empty DataFrame
Columns: []
Index: [spxs, spxu, sds, sh, sso, spxl, upro]

### Use a method to calculate Tracking Error, Mean Difference, and RATD

Here we define a method/function to use for our calculations and then we will be appending those into lists manually. Where we can also add some data we collect from their website such as "Expense Ratio". And of course a column for information of "Inverse/Levered".

In [6]:
def risk_adjusted_tracking_difference(portfolio_returns, benchmark_returns):
    # Calculate tracking error as the standard deviation of the difference in returns
    tracking_error = round(np.std(portfolio_returns - benchmark_returns, ddof=1), 4)

    # Calculate the difference in means of the returns
    mean_difference = round(np.mean(portfolio_returns - benchmark_returns), 4)

    # Calculate the risk-adjusted tracking difference
    ratd = round((mean_difference / tracking_error), 4)
    
    return tracking_error, mean_difference, ratd


In [7]:
risk_adjusted_tracking_difference(spxs['Returns'], spx['Returns'])

(0.0445, -0.0022, -0.0494)

In [8]:
risk_adjusted_tracking_difference(spxu['Returns'], spx['Returns'])

(0.0443, -0.0021, -0.0474)

In [9]:
risk_adjusted_tracking_difference(sds['Returns'], spx['Returns'])

(0.0334, -0.0016, -0.0479)

In [10]:
risk_adjusted_tracking_difference(sh['Returns'], spx['Returns'])

(0.0223, -0.001, -0.0448)

In [11]:
risk_adjusted_tracking_difference(sso['Returns'], spx['Returns'])

(0.0111, 0.0006, 0.0541)

In [12]:
risk_adjusted_tracking_difference(spxl['Returns'], spx['Returns'])

(0.0221, 0.0011, 0.0498)

In [13]:
risk_adjusted_tracking_difference(upro['Returns'], spx['Returns'])

(0.0221, 0.0011, 0.0498)

In [14]:
spx_te_df['Inverse/Levered'] = ['-3x','-3x','-2x','-1x','2x','3x','3x']
spx_te_df['Expense Ratio(%)'] = ['1.08','0.90','0.90','0.89','0.89','1.00','0.91']
spx_te_df['Tracking Error'] = [0.0445, 0.0443, 0.0334, 0.0223, 0.0111, 0.0221, 0.0221]
spx_te_df['Mean(Tracking) Difference'] = [-0.0022,-0.0021,-0.0016,-0.001,0.0006,0.0011,0.0011]
spx_te_df['RATD'] = [-0.0494,-0.0474,-0.0479,-0.0448,0.0541,0.0498,0.0498]

In [15]:
spx_te_df

Inverse/Levered Expense Ratio(%)  Tracking Error  \
funds                                                    
spxs              -3x             1.08          0.0445   
spxu              -3x             0.90          0.0443   
sds               -2x             0.90          0.0334   
sh                -1x             0.89          0.0223   
sso                2x             0.89          0.0111   
spxl               3x             1.00          0.0221   
upro               3x             0.91          0.0221   

       Mean(Tracking) Difference    RATD  
funds                                     
spxs                     -0.0022 -0.0494  
spxu                     -0.0021 -0.0474  
sds                      -0.0016 -0.0479  
sh                       -0.0010 -0.0448  
sso                       0.0006  0.0541  
spxl                      0.0011  0.0498  
upro                      0.0011  0.0498

## Return Summary

We can run regression analysis to get some more data such as beta, t-value, Adj R square. And  later we can append these data into our dataframe.

In [16]:
def run_reg(dep, indep):
    # Create a new DataFrame with the independent variable
    X = indep[['Log Returns']]

    # Add a constant to the independent variable
    X = sm.add_constant(X)

    # Create a new DataFrame with the dependent variable
    y = dep['Log Returns']

    # Fit the linear regression model
    model = sm.OLS(y, X).fit()

    # Print the model summary
    print(model.summary())

In [17]:
run_reg(spxs, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.990
Model:                            OLS   Adj. R-squared:                  0.990
Method:                 Least Squares   F-statistic:                 3.551e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:38:57   Log-Likelihood:                 14898.
No. Observations:                3463   AIC:                        -2.979e+04
Df Residuals:                    3461   BIC:                        -2.978e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0010   5.57e-05    -17.418      

In [18]:
run_reg(spxu, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.989
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                 3.034e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:38:59   Log-Likelihood:                 14652.
No. Observations:                3463   AIC:                        -2.930e+04
Df Residuals:                    3461   BIC:                        -2.929e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0009   5.98e-05    -15.661      

In [19]:
run_reg(sds, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.994
Model:                            OLS   Adj. R-squared:                  0.994
Method:                 Least Squares   F-statistic:                 5.449e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:39:01   Log-Likelihood:                 17048.
No. Observations:                3463   AIC:                        -3.409e+04
Df Residuals:                    3461   BIC:                        -3.408e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0005      3e-05    -16.568      

In [20]:
run_reg(sh, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.994
Model:                            OLS   Adj. R-squared:                  0.994
Method:                 Least Squares   F-statistic:                 5.947e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:39:03   Log-Likelihood:                 19579.
No. Observations:                3463   AIC:                        -3.915e+04
Df Residuals:                    3461   BIC:                        -3.914e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0002   1.44e-05    -12.987      

In [21]:
run_reg(sso, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.997
Model:                            OLS   Adj. R-squared:                  0.997
Method:                 Least Squares   F-statistic:                 1.088e+06
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:39:04   Log-Likelihood:                 18206.
No. Observations:                3463   AIC:                        -3.641e+04
Df Residuals:                    3461   BIC:                        -3.640e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const       -5.177e-05   2.14e-05     -2.414      

In [22]:
run_reg(spxl, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.995
Model:                            OLS   Adj. R-squared:                  0.995
Method:                 Least Squares   F-statistic:                 6.779e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:39:05   Log-Likelihood:                 15973.
No. Observations:                3463   AIC:                        -3.194e+04
Df Residuals:                    3461   BIC:                        -3.193e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0003   4.09e-05     -7.092      

In [23]:
run_reg(upro, spx)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.994
Model:                            OLS   Adj. R-squared:                  0.994
Method:                 Least Squares   F-statistic:                 5.836e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        14:39:07   Log-Likelihood:                 15709.
No. Observations:                3463   AIC:                        -3.141e+04
Df Residuals:                    3461   BIC:                        -3.140e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0003   4.41e-05     -6.469      

### Add found data to our dataframe

Here we append our data to our dataframe manually. 

In [24]:
spx_te_df['Beta'] = [-2.9584,-2.9361,-1.9699,-0.9908,1.9920,2.9970,3.0004]
spx_te_df['t'] = [-595.900,-550.852,-738.174,-771.192,1042.929,823.369,763.956]
spx_te_df['Adj. R sq'] = [0.990,0.989,0.994,0.994,0.997,0.995,0.994]

In [25]:
spx_te_df

Inverse/Levered Expense Ratio(%)  Tracking Error  \
funds                                                    
spxs              -3x             1.08          0.0445   
spxu              -3x             0.90          0.0443   
sds               -2x             0.90          0.0334   
sh                -1x             0.89          0.0223   
sso                2x             0.89          0.0111   
spxl               3x             1.00          0.0221   
upro               3x             0.91          0.0221   

       Mean(Tracking) Difference    RATD    Beta         t  Adj. R sq  
funds                                                                  
spxs                     -0.0022 -0.0494 -2.9584  -595.900      0.990  
spxu                     -0.0021 -0.0474 -2.9361  -550.852      0.989  
sds                      -0.0016 -0.0479 -1.9699  -738.174      0.994  
sh                       -0.0010 -0.0448 -0.9908  -771.192      0.994  
sso                       0.0006  0.0541  1.9920  1042.929      0.997  
spxl                      0.0011  0.0498  2.9970   823.369      0.995  
upro                      0.0011  0.0498  3.0004   763.956      0.994

### SPXS(dependent) and SPX(independent) Regression as an Example on how to read output data of Linear Regression

This output is from a linear regression model. The dependent variable is "Log Returns" and there is one independent variable, which is also "Log Returns". 

The R-squared value of 0.990 indicates that 99% of the variation in the dependent variable is explained by the independent variable. 

The Adjusted R-squared value of 0.990 is similar to the R-squared value and takes into account the number of independent variables in the model. 

The F-statistic tests the overall significance of the regression model, and the probability associated with it is very low (less than 0.05), indicating that the model is significant. 

The coefficients associated with the constant and independent variable show the intercept and the slope of the regression line. 

The P-values associated with the coefficients indicate whether they are statistically significant or not, and in this case, they are both significant at a 5% significance level. 

The standard errors estimate the variability of the estimated coefficients, and the t-values show whether the coefficients are statistically significant or not. 

The confidence intervals provide a range of values within which the true population coefficients are likely to lie. 

The summary also includes additional information such as AIC and BIC, which are measures of the quality of the model fit. 

The Omnibus, Durbin-Watson, Jarque-Bera, and Kurtosis statistics provide additional information about the model assumptions and goodness of fit.

In regression analysis, t-values are used to test the statistical significance of each coefficient estimate. In this case, the t-value for the intercept (const) is -17.418 and the t-value for the Log Returns variable is -595.900. The p-value for both coefficients is very small (p<0.05), indicating that both the intercept and the Log Returns variable are statistically significant.

A t-value measures the number of standard deviations the coefficient is away from zero. In other words, it tells you how statistically significant the coefficient is. The larger the absolute value of the t-value, the more statistically significant the coefficient is. In this case, the t-value for the Log Returns variable is much larger than that of the intercept, indicating that the Log Returns variable is more statistically significant in explaining the variation in the dependent variable.

A negative t-value indicates that the coefficient estimate for the independent variable is negative, i.e., there is a negative relationship between the independent variable and the dependent variable. In the context of regression analysis, the t-value is used to determine the statistical significance of the regression coefficient estimate.

In the case of the regression result, the t-value for the coefficient of "Log Returns" is -595.900, which is quite large. This indicates that the estimated coefficient for "Log Returns" is significantly different from zero and has a strong effect on the dependent variable. The negative sign of the t-value suggests that the coefficient of "Log Returns" is negative and indicates an inverse relationship between the dependent variable and "Log Returns". The sign of the t-value does not necessarily imply the direction of causation between the independent and dependent variables. It only indicates the direction of the relationship between the variables.

In statistical hypothesis testing, the null hypothesis is a statement that there is no significant difference between two variables or no effect of a certain treatment or intervention. On the other hand, the alternative hypothesis is the opposite of the null hypothesis, stating that there is a significant difference or effect.

When a statistical test is conducted, a p-value is calculated, which represents the probability of obtaining the observed results or more extreme results if the null hypothesis is true. If the p-value is less than a predetermined significance level (usually 0.05), we reject the null hypothesis, and if it is greater than the significance level, we fail to reject the null hypothesis.

In the regression output, under the "P>|t|" column, you can see the p-values associated with each coefficient. In this case, the p-value for the intercept is less than 0.05, indicating that it is significantly different from zero. On the other hand, the p-value for the Log Returns coefficient is equal to 0, indicating that it is significantly different from zero with a high level of confidence.

Therefore, we cannot reject the alternative hypothesis that there is a significant relationship between the dependent variable and the independent variable, and the coefficient estimate (-2.9584) is a reliable estimate of the true population coefficient.

## Add HPR column to the DataFrame

Here we create a method to calculate holding period returns of the etfs and then we create a list to append them into our dataframe.

In [26]:
def etf_hpr(etf):
    
    start_value = etf.iloc[0]['Adj Close']
    end_value = etf.iloc[-1]['Adj Close']
    etf_HPR = round(((end_value/start_value)-1)*100, 2)
    
    return etf_HPR

In [27]:
etf_hpr(spxs)

-99.96

In [28]:
etf_hpr(spxu)

-99.95

In [29]:
etf_hpr(sds)

-99.02

In [30]:
etf_hpr(sh)

-87.86

In [31]:
etf_hpr(sso)

1475.66

In [32]:
etf_hpr(spxl)

2939.72

In [33]:
etf_hpr(upro)

3004.23

In [34]:
spx_te_df['HPR %'] = [-99.96, -99.95, -99.02, -87.86, 1475.66, 2939.72, 3004.23]

In [35]:
spx_te_df

Inverse/Levered Expense Ratio(%)  Tracking Error  \
funds                                                    
spxs              -3x             1.08          0.0445   
spxu              -3x             0.90          0.0443   
sds               -2x             0.90          0.0334   
sh                -1x             0.89          0.0223   
sso                2x             0.89          0.0111   
spxl               3x             1.00          0.0221   
upro               3x             0.91          0.0221   

       Mean(Tracking) Difference    RATD    Beta         t  Adj. R sq    HPR %  
funds                                                                           
spxs                     -0.0022 -0.0494 -2.9584  -595.900      0.990   -99.96  
spxu                     -0.0021 -0.0474 -2.9361  -550.852      0.989   -99.95  
sds                      -0.0016 -0.0479 -1.9699  -738.174      0.994   -99.02  
sh                       -0.0010 -0.0448 -0.9908  -771.192      0.994   -87.86  
sso                       0.0006  0.0541  1.9920  1042.929      0.997  1475.66  
spxl                      0.0011  0.0498  2.9970   823.369      0.995  2939.72  
upro                      0.0011  0.0498  3.0004   763.956      0.994  3004.23

This table provides information on the expense ratio, tracking error, mean tracking difference, risk-adjusted tracking difference (RATD), beta, t-value, adjusted R-squared, and HPR% (holding period return) for seven leveraged and inverse ETFs (exchange-traded funds) that are designed to track the performance of the S&P 500 index. The funds are categorized based on their inverse or leveraged exposure, ranging from -3x to 3x.

The data suggests that the leveraged funds (SSO, SPXL, and UPRO) generally have a higher expense ratio and HPR% compared to the inverse funds (SPXS, SPXU, and SDS), but also higher tracking error and mean tracking difference. The -1x fund (SH) has the lowest expense ratio and HPR%, but also a low tracking error and mean tracking difference. The RATD values suggest that the leveraged funds have higher risk-adjusted tracking differences than the inverse funds, except for the -3x funds (SPXS and SPXU) which have negative RATD values indicating better tracking performance. The beta values suggest that the leveraged funds have higher market sensitivity than the inverse funds.

### Russell 2000

In [47]:
rut.shape

(3613, 8)

In [45]:
start_value = rut.iloc[0]['Adj Close']
end_value = rut.iloc[-1]['Adj Close']

benchmark_HPR = (end_value/start_value)-1
benchmark_HPR

3.5894994059008205

Russell 2000 Observeation Number is 3613 and HPR is 358.94% during the period of November 19 - March 31, 2023.

In [54]:
rut_fund_names = ['tza', 'twm', 'rwm', 'uwm', 'tna']

rut_te_df = pd.DataFrame(columns=[])
rut_te_df['funds'] = rut_fund_names
rut_te_df= rut_te_df.set_index('funds')
rut_te_df

Empty DataFrame
Columns: []
Index: [tza, twm, rwm, uwm, tna]

In [55]:
risk_adjusted_tracking_difference(tza['Returns'], rut['Returns'])

(0.0613, -0.0024, -0.0392)

In [56]:
risk_adjusted_tracking_difference(twm['Returns'], rut['Returns'])

(0.0461, -0.0017, -0.0369)

In [57]:
risk_adjusted_tracking_difference(rwm['Returns'], rut['Returns'])

(0.031, -0.0011, -0.0355)

In [58]:
risk_adjusted_tracking_difference(uwm['Returns'], rut['Returns'])

(0.015, 0.0005, 0.0333)

In [59]:
risk_adjusted_tracking_difference(tna['Returns'], rut['Returns'])

(0.0297, 0.001, 0.0337)

In [60]:
rut_te_df['Inverse/Levered'] = ['-3x', '-2x','-1x','2x','3x']
rut_te_df['Expense Ratio(%)'] = ['1.03','0.95','0.95','0.95','1.09']
rut_te_df['Tracking Error'] = [0.0613, 0.0461, 0.031, 0.015, 0.0297]
rut_te_df['Mean(Tracking) Difference'] = [-0.0024, -0.0017, -0.0011, 0.0005, 0.0010]
rut_te_df['RATD'] = [-0.0392, -0.0369, -0.0355, 0.0333, 0.0337]

In [61]:
rut_te_df

Inverse/Levered Expense Ratio(%)  Tracking Error  \
funds                                                    
tza               -3x             1.03          0.0613   
twm               -2x             0.95          0.0461   
rwm               -1x             0.95          0.0310   
uwm                2x             0.95          0.0150   
tna                3x             1.09          0.0297   

       Mean(Tracking) Difference    RATD  
funds                                     
tza                      -0.0024 -0.0392  
twm                      -0.0017 -0.0369  
rwm                      -0.0011 -0.0355  
uwm                       0.0005  0.0333  
tna                       0.0010  0.0337

In [62]:
run_reg(tza, rut)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.983
Model:                            OLS   Adj. R-squared:                  0.983
Method:                 Least Squares   F-statistic:                 2.130e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        15:46:40   Log-Likelihood:                 13417.
No. Observations:                3613   AIC:                        -2.683e+04
Df Residuals:                    3611   BIC:                        -2.682e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0017   9.82e-05    -17.505      

In [63]:
run_reg(twm, rut)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.983
Model:                            OLS   Adj. R-squared:                  0.983
Method:                 Least Squares   F-statistic:                 2.131e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        15:47:53   Log-Likelihood:                 14891.
No. Observations:                3613   AIC:                        -2.978e+04
Df Residuals:                    3611   BIC:                        -2.976e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0009   6.53e-05    -13.778      

In [64]:
run_reg(rwm, rut)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.989
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                 3.214e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        15:48:08   Log-Likelihood:                 18102.
No. Observations:                3613   AIC:                        -3.620e+04
Df Residuals:                    3611   BIC:                        -3.619e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0003   2.69e-05    -12.148      

In [65]:
run_reg(uwm, rut)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.990
Model:                            OLS   Adj. R-squared:                  0.990
Method:                 Least Squares   F-statistic:                 3.445e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        15:48:23   Log-Likelihood:                 15700.
No. Observations:                3613   AIC:                        -3.140e+04
Df Residuals:                    3611   BIC:                        -3.138e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0002   5.22e-05     -3.589      

In [66]:
run_reg(tna, rut)

                            OLS Regression Results                            
Dep. Variable:            Log Returns   R-squared:                       0.988
Model:                            OLS   Adj. R-squared:                  0.988
Method:                 Least Squares   F-statistic:                 2.884e+05
Date:                Sat, 06 May 2023   Prob (F-statistic):               0.00
Time:                        15:48:41   Log-Likelihood:                 13925.
No. Observations:                3613   AIC:                        -2.785e+04
Df Residuals:                    3611   BIC:                        -2.783e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const          -0.0007   8.54e-05     -7.942      

In [67]:
rut_te_df['Beta'] = [-2.8781, -1.9149, -0.9668, 1.9457, 2.9102]
rut_te_df['t'] = [-461.506, -461.613, -566.953, 586.918, 537.011]
rut_te_df['Adj. R sq'] = [0.983, 0.983,  0.989,  0.990,  0.988]

In [68]:
rut_te_df

Inverse/Levered Expense Ratio(%)  Tracking Error  \
funds                                                    
tza               -3x             1.03          0.0613   
twm               -2x             0.95          0.0461   
rwm               -1x             0.95          0.0310   
uwm                2x             0.95          0.0150   
tna                3x             1.09          0.0297   

       Mean(Tracking) Difference    RATD    Beta        t  Adj. R sq  
funds                                                                 
tza                      -0.0024 -0.0392 -2.8781 -461.506      0.983  
twm                      -0.0017 -0.0369 -1.9149 -461.613      0.983  
rwm                      -0.0011 -0.0355 -0.9668 -566.953      0.989  
uwm                       0.0005  0.0333  1.9457  586.918      0.990  
tna                       0.0010  0.0337  2.9102  537.011      0.988

In [69]:
etf_hpr(tza)

-100.0

In [70]:
etf_hpr(twm)

-99.79

In [71]:
etf_hpr(rwm)

-92.94

In [72]:
etf_hpr(uwm)

903.67

In [73]:
etf_hpr(tna)

656.37

In [74]:
rut_te_df['HPR %'] = [-100.0, -99.79, -92.94, 903.67, 656.37]

In [75]:
rut_te_df

Inverse/Levered Expense Ratio(%)  Tracking Error  \
funds                                                    
tza               -3x             1.03          0.0613   
twm               -2x             0.95          0.0461   
rwm               -1x             0.95          0.0310   
uwm                2x             0.95          0.0150   
tna                3x             1.09          0.0297   

       Mean(Tracking) Difference    RATD    Beta        t  Adj. R sq   HPR %  
funds                                                                         
tza                      -0.0024 -0.0392 -2.8781 -461.506      0.983 -100.00  
twm                      -0.0017 -0.0369 -1.9149 -461.613      0.983  -99.79  
rwm                      -0.0011 -0.0355 -0.9668 -566.953      0.989  -92.94  
uwm                       0.0005  0.0333  1.9457  586.918      0.990  903.67  
tna                       0.0010  0.0337  2.9102  537.011      0.988  656.37

The funds in this table are all related to the Russell 2000 index, with varying degrees of leverage. The inverse/leveraged funds in this table are tza (-3x), twm (-2x), uwm (2x), tna (3x), and rwm (-1x). The funds have different expense ratios ranging from 0.95% to 1.09%. Tracking error, which measures the difference between a fund's performance and its benchmark index, varies from 0.0150 to 0.0613. The mean tracking difference, which is the average difference between a fund's daily returns and its benchmark index, ranges from -0.0024 to 0.0010. RATD, which measures a fund's ability to track its benchmark index over time, ranges from -0.0392 to 0.0337. The funds have varying beta values ranging from -2.8781 to 2.9102, and t-statistics ranging from -566.953 to 586.918. The adjusted R-squared value, which measures the percentage of a fund's performance that can be explained by its benchmark index, ranges from 0.983 to 0.990. Finally, the holding period return percentage for these funds ranges from -100.00% to 903.67%.

## Conclusion

The data suggests that leveraged and inverse ETFs can experience significant tracking errors and underperform their stated benchmark over the long term. The ETFs with higher leverage tend to have higher expenses ratios and tracking errors, and the risk-adjusted returns (RATD) tend to be negative for inverse ETFs and positive for leveraged ETFs.

Investors should carefully consider their investment objectives, risk tolerance, and investment horizon before investing in leveraged or inverse ETFs, as these funds can be much riskier and more volatile than traditional ETFs. Additionally, leveraged and inverse ETFs are typically designed for short-term trading and should not be used as long-term investment vehicles.

## Reference

Levered and Inverse Exchange-Traded Products: Blessing or Curse? by Colby J. Pessina and Robert E. Whaley
Published by Financial Analysts Journal | A Publication of CFA Institute https://doi.org/10.1080/0015198X.2020.1830660